Dockerfile/Dockerfile

```diff
@@ -0,0 +1,71 @@
+# Ubuntu Trusty Docker file for base TileDB Dependencies
+
+# This Dockerfile copies files from the parent directory, requiring a build
+# context from the parent directory. From `./..`, execute the following to
+# build:
+# docker build -f Dockerfile/Dockerfile -t tiledb:dev .
+#
+# Use the build arg 'enable' to configure optional TileDB components, e.g.:
+#   docker build -f Dockerfile/Dockerfile --build-arg enable=s3,hdfs -t tiledb:dev
+
+# Ubuntu Trusty
+FROM ubuntu:18.04
+
+# Optional components to enable (defaults to empty).
+ARG enable
+
+# Setup home environment
+RUN useradd tiledb
+RUN mkdir /home/tiledb && chown tiledb /home/tiledb
+ENV HOME /home/tiledb
+
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    wget \
+    unzip \
+    git \
+    cmake \
+    python3 \
+    python3-dev \
+    python3-pip \
+    libssl-dev \
+    && apt-get clean \
+    && apt-get purge -y \
+    && rm -rf /var/lib/apt/lists*
+
+RUN cd /tmp \
+    && wget https://github.com/Kitware/CMake/releases/download/v3.17.2/cmake-3.17.2-Linux-x86_64.tar.gz \
+    && tar -xzf cmake-3.17.2-Linux-x86_64.tar.gz \
+    && cp -R cmake-3.17.2-Linux-x86_64/bin /usr/ \
+    && cp -R cmake-3.17.2-Linux-x86_64/doc /usr/ \
+    && cp -R cmake-3.17.2-Linux-x86_64/man /usr/ \
+    && cp -R cmake-3.17.2-Linux-x86_64/share /usr/
+
+# Copy TileDB source files from the build context to `/home/tiledb/TileDB``
+COPY CMakeLists.txt /home/tiledb/TileDB/CMakeLists.txt
```

Comment:
Is there a reason to copy all the folders and sections and not just add the entire repo directory?

```
COPY . /home/tiledb/TileDB
```

Reply:
Yeah, this was intentional. Mostly to avoid the `build` directory. I figured a whitelist of top-level directories fits better with the philosophy of keeping Dockerfiles as small as possible. It also reduces the likelihood of a future top-level directory sneaking into the image.

Reply:
The `build` directory is created inside the dockerfile a few lines below in the bootstrap and make line right?